src/flamenco/runtime/fd_blockstore.c

```diff
@@ -548,6 +548,64 @@ fd_blockstore_buffered_shreds_remove( fd_blockstore_t * blockstore, ulong slot )
   return FD_BLOCKSTORE_OK;
 }
 
+ulong
+fd_blockstore_block_archive( fd_blockstore_ser_t * ser, int fd, ulong slot ) {
+  ulong off = 0;
+  ulong wsz; 
+  int err = fd_io_write( fd, ser->block_map, sizeof(fd_block_map_t), sizeof(fd_block_map_t), &wsz );
+  off += wsz;
+  err |= fd_io_write( fd, ser->block, sizeof(fd_block_t), sizeof(fd_block_t), &wsz );
+  off += wsz;
+  err |= fd_io_write( fd, ser->data, ser->block->data_sz, ser->block->data_sz, &wsz );
+  off += wsz;
+  if( FD_LIKELY( err == 0 ) ) FD_LOG_DEBUG(( "[%s] archived block %lu", __func__, slot ));
+  else FD_LOG_ERR(( "[%s] failed to archive block %lu", __func__, slot ));
+  return off;
+}
+
+int
+fd_blockstore_block_restore_meta( int fd, 
```

Comment:
naming nit but generally put the verb last eg. `fd_blockstore_block_meta_restore`